pep-0577.rst

```diff
@@ -1,14 +1,41 @@
 PEP: 577
 Title: Augmented Assignment Expressions
 Author: Nick Coghlan <ncoghlan@gmail.com>
-Status: Draft
+Status: Withdrawn
 Type: Standards Track
 Content-Type: text/x-rst
 Created: 14-May-2018
 Python-Version: 3.8
 Post-History: 22-May-2018
 
 
+PEP Withdrawal
+==============
+
+While working on this PEP, I realised that it didn't really address what was
+actually bothering me about PEP 572's proposed scoping rules for previously
+unreferenced assignment targets, and also had some significant undesirable
+consequences (most notably, allowing ``>>=` and ``<<=`` as inline augmented
+assignment operators that meant something entirely different from the ``>=``
+and ``<=`` comparison operators).
+
+I also realised that even without dedicated syntax of their own, PEP 572 allows
+inline augmented assignments to be written using the ``operator`` module::
+
+    from operator import iadd
+    if (target := iadd(target, value)) < limit:
+        ...
+
+(The restriction to simple names as inline assignment targets means that the
+target expession can always be repeated without side effects)
+
+Accordingly, I'm withdrawing this PEP without submitting it for pronouncement,
+and will instead be writing a replacement PEP that focuses specifically on the
+handling of assignment targets which haven't already been declared as local
+variables in the current scope (for both regular block scopes, and for scoped
+expressions).
```

Comment:
Sigh. You keep changing your mind. That's fine, but it seems a burden to bother the PEP process with your personal thought processes to this extent. Can't you just present your next proposal on python-ideas first? Or if it benefits from trying to write about it carefully, write a blog post?

Reply:
Sure, I can do that, but last time I asked Chris whether or not he was interested in a "given" based proposal, he said no (hence drafting a competing PEP in the first place, even though it morphed into something other than the original concept).

I'll still start with a python-ideas post, though - it may be that he'll find this variant of the idea more appealing than the one I asked him about previously.